deps/rabbit/src/rabbit_deprecated_feature_extra.erl

```diff
@@ -0,0 +1,62 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% @copyright 2018-2023 VMware, Inc. or its affiliates.
+%%
+%% @doc
+%% This module provides extra functions unused by the feature flags
+%% subsystem core functionality.
+
+-module(rabbit_deprecated_feature_extra).
+
+-export([cli_info/1]).
+
+-type cli_info() :: [cli_info_entry()].
+%% A list of deprecated feature properties, formatted for the RabbitMQ CLI.
+
+-type cli_info_entry() :: [{name, rabbit_feature_flags:feature_name()} |
+                           {deprecation_phase, rabbit_deprecated_features:deprecation_phase()} |
+                           {provided_by, atom()} |
+                           {desc, string()} |
+                           {doc_url, string()}].
+%% A list of properties for a single deprecated feature, formatted for the
+%% RabbitMQ CLI.
+
+-spec cli_info(Which :: all | used) -> cli_info().
```

Comment:
Could you please name arguments and the return value, then define them below? See `rabbit_deprecated_features` for some examples.

Reply:
They're just defined above.

Reply:
No, I mean, something like that:
```erlang
-spec cli_info(Which) -> CliInfo when
    Which :: all | used,
    CliInfo :: cli_info().
```

I find the spec more readable like this: the prototype uses named arguments and return value to get the meaning of everything, then you have the type for more details.